src/dotnet/CarbonAware.DataSources.Json/JsonDataSource.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.Diagnostics;
+using System.Reflection;
+using CarbonAware.Interfaces;
+using CarbonAware.Model;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace CarbonAware.DataSources.Json;
+
+/// <summary>
+/// Reprsents a JSON data source.
+/// </summary>
+public class JsonDataSource : ICarbonIntensityDataSource
+{
+    public string Name => "JsonDataSource";
+
+    public string Description => "Example plugin to read data from a json for Carbon Aware SDK";
+
+    public string Author => "Microsoft";
+
+    public string Version => "0.0.1";
+
+    private List<EmissionsData>? emissionsData;
+
+    private readonly ILogger<JsonDataSource> _logger;
+
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="JsonDataSource"/> class.
+    /// </summary>
+    /// <param name="logger">The logger for the datasource</param>
+    public JsonDataSource(ILogger<JsonDataSource> logger )
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    public Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(IEnumerable<Location> locations, DateTimeOffset startPeriod, DateTimeOffset endPeriod)
```

Comment:
The names of these params changed in the interface as one of the last commits before it merged.  Let's update this to match.

https://github.com/microsoft/carbon-aware-sdk/pull/26/files#diff-bf5399ab973a52b25da9fb0a4d2dc033cf304f755b9354cd98e578c4d2238f43R20